devdoc/LDAPSecurityManager.md

```diff
@@ -0,0 +1,92 @@
+types. Additionally, defined DNs may be adjusted to match AD type requirements.
+
+# LDAP Operations
+This section covers what it does....
+
+## Connections
+The manager handles connections to an AD in order to perform required operations. It is possible that values such
+as permissions and resources may be cached to improve performance.
+
+### Connection Pool
+The LDAP security manager maintains a pool of LDAP connections. The pool is limited in size to _maxConnections_ 
+from the configuration. The connection pool starts empty. As connections are created, each is added to the pool
+until the max allowed is reached. 
+
+The following process is used when an LDAP connection is needed.
+
+First, the connection pool is searched for a free connection. If found and valid, the connection is returned. A 
+connection is considered free if no one is using it and valid if the AD can be accessed. If no valid free connections 
+are found, a new uninitialized connection is created.
+
+For a new connection, an attempt is made to connect to each AD starting with the current. See **Handling AD Hosts**
+below for how ADs are cycled when a connection fails.For each AD, as it cycles through, connection attempts are retried 
+with a short delay between each. If unable to connect, the AD host is marked rejected and the next is attempted. Once a
+new connection has been established, if the max number of connections has not been reached yet, the connection is added 
+to the pool. 
+
+It is important to note that if the pool has reached its max size, new connections will continue to be made, but
+are not saved in the pool. This allows the pool to maintain a steady working state, but allow for higher demand. 
+Connections not saved to the pool are deleted once no longer in use.
+
+### Handling AD Hosts
+The manager keeps a list of AD hosts and the index of the current host. The current host is used for all AD
+operations until there is a failure. At that time the manager marks the host as "rejected" and moves to the 
+next host using a round-robin scheme. 
+
+## User Authentication & Authorization
+This is the process of verifying the identity of a user. It also verifies authorization for an optional resource list.
+The security manager _authorize_ interface method performs this task. It accepts a user and optional resource list.
+Currently only a username and password are required to authenticating the user. Once authenticated, the user is 
+checked to make sure they are authorized to access the optional resources listed. As an example, users logging into
+ECLWatch require **SmcAccess** permission. It is possible a user authenticates, but is not authorized for one or more
+optional resources. In this case, the _authorize_ call will fail.
+
+The LDAP security manager may cache user credentials in order to speed up authentication (not resource authorization)
+when a user attempts multiple authorization calls. The cache is 
+
```

Comment:
"The cache is" _____________.
Missing the rest of the sentence

Reply:
This last section should have been removed as well. It is now covered in another document.